src/containers/BooksForm.js

```diff
@@ -11,15 +14,81 @@ const CATEGORIES = [
   'Sci-Fi',
 ];
 
-const handleFormSubmit = () => {
-};
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: CATEGORIES[0],
+    };
+
+    this.handleSelect = this.handleSelect.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+  }
+
+
+  handleInputChange(e) {
+    this.setState({ title: e.target.value });
+  }
+
+  handleSelect(e) {
+    this.setState({ category: e.target.value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const { title, category } = this.state;
+    const { createBook } = this.props;
+    const book = {
+      id: idGen(),
+      title,
+      category,
+    };
+    this.setState({
+      title: '',
+      category: CATEGORIES[0],
+    });
+    createBook(book);
+  }
 
-const BooksForm = () => (
-  <Form
-    placeholder="Book Title"
-    options={CATEGORIES}
-    submitHandler={handleFormSubmit}
-  />
-);
+  render() {
+    const { title, category } = this.state;
+
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <input
+          type="text"
+          placeholder="Book Title"
+          value={title}
+          onChange={this.handleInputChange}
+        />
+
+        <select
+          value={category}
+          onChange={this.handleSelect}
+        >
+          {CATEGORIES.map(c => (
+            <option
+              key={c}
+              value={c}
+            >
+              {c}
+            </option>
+          ))}
+        </select>
+        <input type="submit" value="Save the Book" />
+      </form>
+    );
+  }
+}
+
+const mapDispatchToProps = dispatch => ({
+  createBook(book) { dispatch(createBook(book)); },
+});
```

Comment:
[Optional] We can define mapStateToProps simpler as an object ([docs](https://react-redux.js.org/using-react-redux/connect-mapdispatch#defining-mapdispatchtoprops-as-an-object)):
```
const mapDispatchToProps = { createBook };
```

Reply:
Thanks for the suggestion. It is really handy and nice.